Xamarin.Essentials/TextToSpeech/TextToSpeech.tizen.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Tizen.Uix.Tts;
+
+namespace Xamarin.Essentials
+{
+    public static partial class TextToSpeech
+    {
+        static TtsClient tts = null;
+        static TaskCompletionSource<bool> tcsInitialize = null;
+        static TaskCompletionSource<bool> tcsUtterances = null;
+
+        internal static async Task PlatformSpeakAsync(string text, SpeechOptions settings, CancellationToken cancelToken = default)
+        {
+            await Initialize();
+
+            if (tcsUtterances?.Task != null)
+                await tcsUtterances.Task;
+
+            tcsUtterances = new TaskCompletionSource<bool>();
+            if (cancelToken != null)
+            {
+                cancelToken.Register(() =>
+                {
+                    tts?.Stop();
+                    tcsUtterances?.TrySetResult(true);
+                });
+            }
+
+            var language = "en_US";
+            var voiceType = Voice.Auto;
+            if (settings?.Locale.Language != null)
+            {
+                foreach (var voice in tts.GetSupportedVoices())
+                {
+                    if (voice.Language == settings.Locale.Language)
+                    {
+                        language = voice.Language;
+                        voiceType = voice.VoiceType;
+                    }
+                }
+            }
+            var speed = 0;
+            if (settings?.Pitch.HasValue ?? false)
+                speed = (int)settings.Pitch.Value;
```

Comment:
Should there be a conversion of some sort here for Pitch?  We treat pitch as a minimum of `0.0f`, normal is `1.0f` and fastest is `2.0f`.

Reply:
the default value of speed is `0` (it means `auto`) in tizen. So, we'd like to set speed only if pitch value is specified by user. (like android implementation as below)

https://github.com/xamarin/Essentials/blob/05ba0792df416f38df998c26488ba92c3f5c90ab/Xamarin.Essentials/TextToSpeech/TextToSpeech.android.cs#L143-L147

Reply:
It's important that the values that will be set in the cross platform API for essentials translate into the correct _relative_ values in the native SDK.  Here's what I'm thinking might work for Tizen in this case:

```csharp
var TIZEN_MAX = 100;
var TIZEN_MIN = -100;
var TIZEN_DEFAULT = 0;

var ESSENTIALS_MAX = 2.0f;
var ESSENTIALS_MIN = 0.0f;
var ESSENTIALS_DEFAULT = 1.0f;

var tizenPitch = 0;

if (essentialsPitch < 0f)
    essentialsPitch = 0f;
if (essentialsPitch > 2.0f)
    essentialsPitch = 2.0f;

if (essentialsPitch < ESSENTIALS_DEFAULT)
    tizenPitch = (ESSENTIALS_MIN + essentialsPitch) * TIZEN_MIN;
else if (essentialsPitch > ESSENTIALS_DEFAULT)
    tizenPitch = (ESSENTIALS_MAX - essentialsPitch) * TIZEN_MAX;
```

I'm not sure what the real min/max are for Tizen's values but that code should roughly do the right thing, where in the end you'll want to set `tts.SetPitch(tizenPitch)`

Reply:
Agreed. (`min` is `1`, `max` is `15` for pitch on tizen for TTS. ).